core/src/main/java/com/netflix/conductor/core/execution/tasks/IsolatedTaskQueueProducer.java

```diff
@@ -0,0 +1,105 @@
+package com.netflix.conductor.core.execution.tasks;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.conductor.common.metadata.tasks.TaskDef;
+import com.netflix.conductor.core.config.Configuration;
+import com.netflix.conductor.core.utils.QueueUtils;
+import com.netflix.conductor.service.MetadataService;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Singleton
+public class IsolatedTaskQueueProducer {
+
+	private static Logger logger = LoggerFactory.getLogger(IsolatedTaskQueueProducer.class);
+	private MetadataService metadataService;
+	private Configuration config;
+	private int pollingTimeOut;
+
+
+	@Inject
+	public IsolatedTaskQueueProducer(MetadataService metadataService, Configuration configuration) {
+		this.metadataService = metadataService;
```

Comment:
I don't see metadataService is used anywhere.

Reply:
Please check line 73. Its used there.